src/lib/cache.ts

```diff
@@ -0,0 +1,35 @@
+import { createClient } from 'redis';
+import { generateRedisConfig } from './queue';
+
+interface CacheOptions {
+  EX?: number;
+}
+
+const {
+  uri: redisUrl,
+  tlsEnabled,
+} = generateRedisConfig();
+
+const redisClient = createClient({
+  url: redisUrl,
+  socket: {
+    tls: tlsEnabled,
+  },
+});
+
```

Comment:
to implement this function, you need to pass it a key and a callback.
The key is the lookup value. The callback is whatever function you need to call to refresh the cache. It needs to do whatever it's doing but return a string, because that is what can be stored in the redis cache. So functions that call this would probably need to call JSON.stringify / JSON.decode on their own